aclk/https_client.c

```diff
@@ -183,18 +183,19 @@ static int process_chunked_content(rbuf_t buf, http_parse_ctx *parse_ctx)
                 char buf_size[HTTP_HDR_BUFFER_SIZE];
                 rbuf_pop(buf, buf_size, idx);
                 buf_size[idx] = 0;
-                parse_ctx->chunk_size = strtol(buf_size, NULL, 16);
-                if (parse_ctx->chunk_size < 0 || parse_ctx->chunk_size == LONG_MAX) {
+                long chunk_size = strtol(buf_size, NULL, 16);
+                if (chunk_size < 0 || chunk_size == LONG_MAX) {
                     netdata_log_error("Chunk size out of range");
                     return PARSE_ERROR;
                 }
+                parse_ctx->chunk_size = chunk_size;
                 if (parse_ctx->chunk_size == 0) {
                     if (errno == EINVAL) {
                         netdata_log_error("Invalid chunk size");
                         return PARSE_ERROR;
                     }
                     parse_ctx->chunked_content_state = CHUNKED_CONTENT_CHUNK_END_CRLF;
-                    return 0;
+                    continue;
```

Comment:
@underhood Is this change intentional? You want to continue instead of returning in this case?

Reply:
intentional change, continue will cause switch to be evaluated again going to `CHUNKED_CONTENT_CHUNK_END_CRLF` case, where we do
```
            case CHUNKED_CONTENT_CHUNK_END_CRLF:
                if (rbuf_bytes_available(buf) < strlen(HTTP_LINE_TERM))
                    return NEED_MORE_DATA;
```

so in this case we avoid `process_chunked_content` to be called again if there is still data in buffer that can be parsed.

I noticed this when fixing CID 398318
I should have made it a separate commit to be more clear

Reply:
previously we did `return 0;` *(0==NEED_MORE_DATA)* even if maybe there is more data in the buffer.

Reply:
`process_chunked_content` should return only if
a) parsing is finished and nothing more is to be expected
b) there was an error parsing
c) we didnt finish parsing but there are no more data in buffer (in which case we go to poll to wait for more data)